pyspectral/near_infrared_reflectance.py

```diff
@@ -195,15 +200,25 @@ def reflectance_from_tbs(self, sun_zenith, tb_near_ir, tb_thermal, **kwargs):
                      absorption correction will be applied.
 
         """
+        # Check for dask arrays
+        if hasattr(tb_near_ir, 'compute') or hasattr(tb_thermal, 'compute'):
+            compute = False
+        else:
+            compute = True
+        if hasattr(tb_near_ir, 'mask') or hasattr(tb_thermal, 'mask'):
+            is_masked = True
+        else:
+            is_masked = False
+
         if np.isscalar(tb_near_ir):
             tb_nir = np.array([tb_near_ir, ])
         else:
-            tb_nir = np.array(tb_near_ir)
+            tb_nir = np.asanyarray(tb_near_ir)
```

Comment:
Does `np.asanyarray` convert dask arrays to numpy arrays?

Reply:
It does:

```
In [1]: import dask.array as da                                                                                                                                      

In [2]: a = da.arange(5., chunks=2)                                                                                                                                  

In [3]: import numpy as np                                                                                                                                           

In [4]: type(np.asanyarray(a))                                                                                                                                       
Out[4]: numpy.ndarray
```

Reply:
@djhoese what do you suggest?

Reply:
As mentioned on slack, I'd prefer if everything could be kept as dask arrays for as long as possible and if things have to be computed they should be done in a `map_blocks` or `delayed` function. As @pnuu mentioned on slack converting to numpy here made the code faster and use less memory in his test cases. I'm fairly certain this is because the array is used later on in a non-dask-friendly way and the dask array is actually being computed multiple times. I think I pointed them out on slack during the PCW.

Reply:
As this part of pyspectral wasn't dask friendly before, could we perhaps merge this PR now and make sure things works and then work further to improve the daskfriendliness?
@mraspaud @djhoese @pnuu ?

Reply:
I'm fine with merging. Then we could also merge https://github.com/pytroll/satpy/pull/529 so this PR would be usable via SatpY.

Reply:
So this is working now ?

Reply:
I think there hasn't been anything that didn't work since I last touched this, just that it might not be quite optimal in performance and how things should be done with dask.

Reply:
Fine. +0

Reply:
Fine, I will just try do some system testing off line first to be sure. Any suggested recipes/tests @pnuu ?